aesthetic_ants/penguin/tile_layer.py

```diff
@@ -18,12 +45,33 @@
 }
 
 
+# Maps decayable tiles to their descendant tile type
+DECAY_MAP = {
+    TileType.ICE: TileType.WEAK_ICE,
+    TileType.WEAK_ICE: TileType.WATER
+}
+
+# Tiles that are weak enough that nearby tiles decay faster
+WEAK_TILES = {TileType.WEAK_ICE, TileType.WATER}
+
+
 class Tile(pyglet.sprite.Sprite):
     """
     Represents an individual tile in a tiled layer
     """
-    def __init__(self, tile: TileType, *args, **kwargs):
+    def __init__(self, tile: TileType, *args, health=0, **kwargs):
         self.tile_type = tile
+
+        if not health:
+            if self.tile_type == TileType.ICE:
+                self.health = random.random() * ICE_HEALTH
+            elif self.tile_type == TileType.WEAK_ICE:
+                self.health = random.random() * WEAK_ICE_HEALTH
+            else:
+                self.health = -1
+        else:
+            self.health = health
+
```

Comment:
It might be a bit big to change (so you don't have to change it if it's very difficult), but would there be a way to do using subclasses? This is a bit wonky.

We can keep the enums though.